### PR TITLE
Feature/placement tool

### PR DESCRIPTION
**Pull request by thecadams**

### CODE REVIEW

**Comment by thecadams** [comment]
@maxb94 wondering if we can work together when you're available to finish this PR and get it merged, only a few small tweaks left for it I think.

File: functions/venue.js [hunk 1]
```diff
@@ -106,6 +106,10 @@ const checkUserIsOwner = async (venueId, uid) => {
     });
 };
 
+const checkUserIsPlayaOwner = async (uid) => {
```
Comment (maxb94) [inline functions/venue.js:109]: not async

File: functions/venue.js [hunk 2]
```diff
@@ -263,3 +267,32 @@ exports.deleteVenue = functions.https.onCall(async (data, context) => {
 
   admin.firestore().collection("venues").doc(venueId).delete();
 });
+
+exports.adminUpdatePlacement = functions.https.onCall(async (data, context) => {
+  console.log("DEBUG: adminUpdatePlacement", data, context.auth.token.user_id);
+  const venueId = data.id;
+  checkAuth(context);
+
+  checkUserIsPlayaOwner(context.auth.token.user_id);
```
Comment (maxb94) [inline functions/venue.js:276]: await

File: src/pages/Admin/Admin.scss [hunk 3]
```diff
@@ -214,6 +221,28 @@ $page-max-width: 1240px;
           padding-right: 40px;
           padding-top: 20px;
         }
+        .page-container-adminpanel-sidebar {
+          background-image: $admin-gradient;
+          overflow: hidden scroll;
```
Comment (maxb94) [inline src/pages/Admin/Admin.scss:226]: is this valid css?

File: src/pages/Admin/PlacementComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+  useFirestoreConnect({
+    collection: "venues",
+    storeAs: "playaVenues",
+  });
+
+  const venues = useSelector((state) => state.firestore.ordered.playaVenues);
+  const venue = venues?.find((venue) => venue.id === venueId);
+  const unplacedVenues = venues?.filter(isUnplaced);
+  const selfPlacedVenues = venues?.filter(isSelfPlaced);
+  const placedVenues = venues?.filter(isPlaced);
```
Comment (maxb94) [inline src/pages/Admin/PlacementComponent.tsx:96]: these all produce new arrays on every pass. Consider memoising.

**Review by maxb94 — APPROVED** [review]

**Pull request merged**